ovos_plugin_manager/templates/language.py

```diff
@@ -46,3 +56,13 @@ def translate_list(self, data, lang_tgt, lang_src="en"):
             elif isinstance(v, list):
                 data[idx] = self.translate_list(v, lang_tgt, lang_src)
         return data
+
+    @property
+    def available_languages(self) -> set:
+        """Return languages supported by this translator implementation in this state
+        This property should be overridden by the derived class to advertise
+        what languages that engine supports.
```

Comment:
what is a supported language here? in translations usually this is defined as language pairs, is this a source or a target language that we are supporting? is it bidirectional? this will all depend on the plugin...

Reply:
Hmm, this might be trickier to define then. I was looking at this as a method to wrap LibreTranslate (https://libretranslate.2022.us/docs/#/translate/post_languages) which just returns a list.

Reply:
so is the assumption that any combo of 2 langs returned by this method can be translated from/to?

Reply:
That's my working assumption; I updated the docstring to state as much.. It couldn't be a property, but we could add or modify this to accept a language arg (default to `self.internal_language`) and return language pairs including that language?

Reply:
i think i like that approach, specify source language and return valid output languages, should cover all use cases across plugins

Reply:
we can keep the property also f you think its useful or only for consistency across plugins, as long as we document both it's fine.

seems redundant here but i like the consistency, a base api we know is available in every plugin would be nice